src/analyzer/analyzerthread.cpp

```diff
@@ -311,11 +311,12 @@ AnalyzerThread::AnalysisResult AnalyzerThread::analyzeAudioSource(
             const double frameProgress =
                     double(audioSource->frameLength() - remainingFrameRange.length()) /
                     double(audioSource->frameLength());
+            // math_min is required to compensate rounding errors
             const AnalyzerProgress progress =
-                    frameProgress *
-                    (kAnalyzerProgressFinalizing - kAnalyzerProgressNone);
+                    math_min(kAnalyzerProgressFinalizing,
+                            frameProgress *
+                                    (kAnalyzerProgressFinalizing - kAnalyzerProgressNone));
```

Comment:
This becomes wrong if kAnalyzerProgressNone is not 0. We should either remove this from formula or fix it. 
I would also not be upset if we remove kAnalyzerProgressNone completely and replace it with a hard-coded 0.   
This makes the code much more readable.

Reply:
Wrong. kAnalyzerProgressNone is the origin and reported progress is supposed to be > kAnalyzerProgressNone. Replacing the constant with a literal doesn't help in any way.

Reply:
Imagine kAnalyzerProgressNone = 0.01
min(0,95, 1 * (0.95 - 0,01) = 0,94 
the value will never reach  kAnalyzerProgressFinalizing

I think we need something like 
(frameProgress * (kAnalyzerProgressFinalizing - kAnalyzerProgressNone) + kAnalyzerProgressNone;

But cluttering the code like that feels stupid for no reason. 
I would just dispose kAnalyzerProgressNone;

Reply:
Out of scope for this PR.

Reply:
Yes, but why not just pick one solution and forget.

Reply:
This is not the cause for the debug assertion.